src/exporter/ceph_exporter.cc

```diff
@@ -64,8 +85,31 @@ int main(int argc, char **argv) {
   }
   common_init_finish(g_ceph_context);
 
+  // Start the web server thread
   boost::thread server_thread(web_server_thread_entrypoint);
-  DaemonMetricCollector &collector = collector_instance();
-  collector.main();
+
+  // Start the DaemonMetricCollector thread
+  boost::thread collector_thread(collector_thread_entrypoint);
+
+  // Main loop that waits for stop signals
+  while (!stop_signal_received.load()) {
+      std::this_thread::sleep_for(std::chrono::milliseconds(500));
+  }
+
+  unregister_async_signal_handler(SIGHUP, sighup_handler);
+  unregister_async_signal_handler(SIGINT, handle_signal);
+  unregister_async_signal_handler(SIGTERM, handle_signal);
+  shutdown_async_signal_handler();
+
+  // Stop the server and collector threads by interrupting them
+  stop_web_server();
+  server_thread.interrupt();  // Interrupt the web server thread
```

Comment:
In the absence of a matching `boost::this_thread::interruption_point()` call, does this line actually do anything?